src/main/java/org/jabref/gui/bibtexkeypattern/BibtexKeyPatternPanel.java

```diff
@@ -72,24 +62,52 @@ private void buildGUI() {
             mode = Globals.prefs.getDefaultBibDatabaseMode();
         }
 
-        int rowIndex = 3;
-        for (EntryType type : EntryTypes.getAllValues(mode)) {
+        int rowIndex = 1;
+        int columnIndex = 0;
+        // The header - can be removed
+        for (int i = 0; i < COLUMNS; i++) {
+            Label label = new Label(Localization.lang("Entry type"));
+            Label keyPattern = new Label(Localization.lang("Key pattern"));
+            gridPane.add(label, ++columnIndex, rowIndex);
+            gridPane.add(keyPattern, ++columnIndex, rowIndex);
+            ++columnIndex; //3
+        }
+
+        rowIndex++;
+        Label defaultPattern = new Label(Localization.lang("Default pattern"));
+        Button button = new Button("Default");
+        button.setOnAction(e-> defaultPat.setText((String) Globals.prefs.defaults.get(JabRefPreferences.DEFAULT_BIBTEX_KEY_PATTERN)));
+        gridPane.add(defaultPattern, 1, rowIndex);
+        gridPane.add(defaultPat, 2, rowIndex);
+        gridPane.add(button, 3, rowIndex);
+
+        Object[] entryTypes = EntryTypes.getAllValues(mode).toArray();
```

Comment:
You can also iterate over the list, you have to use the get(i) methods then. It has the additional advantage that you are still type safe with generics

Reply:
Do you mean something like this?
`List<EntryType> entryTypes = (List<EntryType>) EntryTypes.getAllValues(mode);`